src/lib/transaction_snark/test/util.ml

```diff
@@ -676,24 +678,28 @@ let test_transaction_union ?expected_failure ?txn_global_slot ledger txn =
           ~constraint_constants txn
         |> Or_error.ok_exn )
   in
-  match
-    Or_error.try_with (fun () ->
-        Transaction_snark.check_transaction ~constraint_constants ~sok_message
-          ~source_first_pass_ledger ~target_first_pass_ledger
-          ~init_stack:pending_coinbase_stack
-          ~pending_coinbase_stack_state:
-            { Transaction_snark.Pending_coinbase_stack_state.source =
-                pending_coinbase_stack
-            ; target = pending_coinbase_stack_target
-            }
-          ~supply_increase
-          { transaction = txn; block_data = state_body; global_slot }
-          (unstage @@ Sparse_ledger.handler sparse_ledger) )
-  with
-  | Error _e ->
-      assert expect_snark_failure
-  | Ok _ ->
```

Comment:
This is a really sneaky one. In @cjjdespres 's original PR we're missing a sig kind. But try_with won't tell you anything and silently assume we want to calculate an lambda!

Reply:
Hate such things, good catch. How were you able to spot it @glyh ?